### PR TITLE
Add Linear workflow config and validation

### PR DESCRIPTION
**Pull request by jessmartin**
## Summary
- add a discriminated tracker config union with workflow parsing and validation for `tracker.kind: linear`
- apply upstream Linear defaults and env-backed token/assignee resolution at the config boundary
- document the Linear workflow snippet and add tests plus a narrow runtime guard until the Linear adapter exists

## Testing
- pnpm typecheck
- pnpm lint
- pnpm test

Refs #65

### CODE REVIEW

**Comment by jessmartin** [comment]
Addressed Greptile feedback in `b2a4502`:

- `SUPPORTED_TRACKER_KINDS` is now the authoritative validation source in `resolveTrackerKind`, so the supported-kind list can no longer drift from the error message.
- I kept the env-fallback behavior in `resolveEnvBackedSecret` intentionally. It matches the upstream Elixir `resolve_env_value/2` seam for Linear secrets, and `tests/unit/workflow.test.ts` covers the fallback path explicitly (`$MISSING_LINEAR_SECRET` falling back to `LINEAR_API_KEY`). Changing that behavior in this config-only slice would diverge from the reference contract.

File: src/config/workflow.ts [hunk 1]
```diff
@@ -46,13 +60,76 @@ function requireNumber(value: unknown, field: string): number {
...
+  const trimmed = value.trim();
+  const referencedEnvName = resolveEnvReferenceName(trimmed);
+  if (referencedEnvName === null) {
+    return normalizeSecretValue(trimmed);
+  }
+
+  const referencedValue = process.env[referencedEnvName];
+  if (referencedValue === undefined) {
+    return normalizeSecretValue(process.env[fallbackEnvName]);
+  }
```
[the remark this answers is not in the document]
Comment (jessmartin) [reply src/config/workflow.ts:165]: Keeping this behavior is intentional for this slice. The TypeScript loader is mirroring the upstream Elixir `resolve_env_value/2` contract here: an explicit `$ENV_VAR` reference that is unset falls back to the field-specific default env var (`LINEAR_API_KEY` / `LINEAR_ASSIGNEE`). That behavior is exercised in `tests/unit/workflow.test.ts`, and changing it in this config-only issue would diverge from the reference seam before the Linear adapter lands.

File: src/config/workflow.ts [hunk 2]
```diff
@@ -180,6 +233,110 @@ function resolveConfig(raw: RawWorkflow, workflowPath: string): ResolvedConfig {
...
+    throw new ConfigError("Expected non-empty string for tracker.kind");
+  }
+
+  const normalizedKind = value.trim();
+  if (normalizedKind === "github-bootstrap" || normalizedKind === "linear") {
+    return normalizedKind;
+  }
+
+  throw new ConfigError(
+    `Unsupported tracker.kind '${normalizedKind}'. Supported kinds: ${SUPPORTED_TRACKER_KINDS.join(", ")}`,
```
[the remark this answers is not in the document]
Comment (jessmartin) [reply src/config/workflow.ts]: Addressed in `b2a4502`: `resolveTrackerKind` now delegates the membership check to `SUPPORTED_TRACKER_KINDS`, so the accepted-kind list and the error message share the same source of truth.

**Comment by jessmartin** [comment]
Addressed the remaining Linear config follow-up in `6b7198d`:

- `resolveTrackerConfig` now uses an exhaustive `switch` and `tracker.kind` only defaults to `github-bootstrap` when the key is absent, not when it is explicitly blank.
- Linear `active_states` / `terminal_states` now reject empty arrays at the config boundary.
- `tracker.assignee` no longer inherits `LINEAR_ASSIGNEE` implicitly; env-backed assignee routing remains opt-in via an explicit workflow value like `$LINEAR_ASSIGNEE`.
- Added unit coverage for blank `tracker.kind`, empty Linear state lists, and omitted-assignee ambient env behavior.

Local gates rerun after the patch: `pnpm typecheck`, `pnpm lint`, `pnpm test`, and `codex review --base origin/main`.

**Comment by jessmartin** [comment]
Addressed the remaining Linear config follow-up in `8a6e9c6`:

- `tracker.endpoint` now validates as a real URL at the config boundary, so malformed Linear endpoints fail during workflow load instead of at first network use.
- Added explicit coverage for `tracker.assignee: $UNSET_VAR`, which continues to normalize to `null` rather than inheriting an ambient assignee.
- Fixed the self-review follow-up from `codex review`: prompt rendering now redacts the Linear API key, and `symphony status` / `symphony-report issue` now read only `workspace.root` from `WORKFLOW.md`, so those local inspection commands do not require Linear credentials.

Local gates rerun after the patch: `pnpm typecheck`, `pnpm lint`, `pnpm test`, and `codex review --base origin/main`. I'm watching PR `#72` for the new CI and automated review results.

**Comment by jessmartin** [comment]
Follow-up on `ac7befe`: CI failed only on `pnpm format:check`, so I ran Prettier on the two flagged files (`src/config/workflow.ts` and `tests/unit/cli.test.ts`), reran `pnpm format:check`, `pnpm typecheck`, `pnpm lint`, and `pnpm test`, and pushed the formatting-only fix. I'm watching the restarted checks on the new head.

**Comment by jessmartin** [comment]
Addressed the remaining Greptile follow-up in `5113eed`:

- `tracker.endpoint` validation now rejects non-HTTP(S) schemes, so syntactically valid but unusable endpoints like `ftp://...` fail at workflow-load time.
- Added a unit regression covering the unsupported-scheme case alongside the existing malformed-URL coverage.

Local gates rerun after the patch: `pnpm format:check`, `pnpm typecheck`, `pnpm lint`, and `pnpm test`. I'm watching the restarted PR checks and automated review results on this new head.

**Comment by jessmartin** [comment]
Addressed the new Greptile follow-up in `8ccbcc9`:

- `redactPromptConfig` now delegates to an exhaustive `switch` over `TrackerConfig`, so future tracker kinds cannot silently bypass secret redaction.
- The Linear redaction behavior is unchanged today; this is a compile-time guardrail for the next tracker addition.

Local gates rerun after the patch: `pnpm format:check`, `pnpm typecheck`, `pnpm lint`, and `pnpm test`. I'm watching the refreshed automated review/check state on the new head.

**Comment by jessmartin** [comment]
Addressed the remaining Greptile test-harness follow-up in `c372a86`:

- moved the Linear env mutations in `tests/unit/workflow.test.ts` inside the `try` block so `finally` always restores `LINEAR_API_KEY` / worker env even if fixture setup fails
- applied the same cleanup ordering to the related Linear status/report tests so env restoration is consistent across the CLI/report harnesses
- no production code changed in this patch; it is a test isolation fix only

Local gates rerun after the patch: `pnpm format:check`, `pnpm typecheck`, `pnpm lint`, and `pnpm test`. I also started `codex review --base origin/main`, but the local review CLI did not return a result before timing out in this workspace; the code/test gates above completed successfully. I'm watching the refreshed PR checks and automated review state on this new head.

**Comment by jessmartin** [comment]
Addressed the remaining follow-up in `2bec7fd`:

- `resolveTrackerConfig` now mirrors `redactTrackerConfig` with a runtime `default` branch that delegates to `exhaustiveTrackerConfig`, so the tracker resolver has the same defense-in-depth against unexpected runtime kinds.
- moved the Linear CLI test workflow fixture writes inside the `try` blocks, so temp directories are cleaned even if fixture setup throws before the assertions run.

Local gates rerun after the patch: `pnpm format:check`, `pnpm typecheck`, `pnpm lint`, and `pnpm test`. I also reran `codex review --base origin/main`, but the local review CLI again stalled in diff analysis and did not return findings in this workspace.

File: src/config/workflow.ts [hunk 3]
```diff
@@ -32,27 +47,148 @@ interface PromptRenderInput {
...
 
+function requireObject(
+  value: unknown,
+  field: string,
+): Readonly<Record<string, unknown>> {
+  if (value === undefined) {
+    return {};
+  }
+  if (value === null || typeof value !== "object" || Array.isArray(value)) {
+    throw new ConfigError(`Expected object for ${field}`);
```
[the remark this answers is not in the document]
Comment (jessmartin) [reply src/config/workflow.ts:95]: Addressed in `7f435b7`:\n\n- renamed `requireObject` to `coerceOptionalObject` so the helper name matches its actual `undefined -> {}` contract\n- added an inline comment documenting the intentional asymmetry: omitted top-level sections preserve the legacy coercion path, while explicit YAML `null` now fails early at the config boundary\n- added a unit regression that locks the explicit-null behavior on `tracker` (`Expected object for tracker`) so this stricter edge-case handling stays intentional\n\nLocal gates rerun after the patch: `pnpm format:check`, `pnpm typecheck`, `pnpm lint`, and `pnpm test`. I also retried `codex review --base origin/main`, but the local review CLI stalled twice in diff analysis in this workspace before returning findings.

File: src/tracker/factory.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+export function createTracker(config: TrackerConfig, logger: Logger): Tracker {
+  switch (config.kind) {
+    case "github-bootstrap":
+      return new GitHubBootstrapTracker(config, logger);
+    case "linear":
+      throw new Error(
+        "tracker.kind 'linear' is not yet supported by `symphony run`; workflow loading and validation are available, but the Linear tracker adapter has not been implemented yet.",
+      );
+  }
```
[the remark this answers is not in the document]
Comment (jessmartin) [reply src/tracker/factory.ts:17]: Addressed in `a3eba83`:\n\n- `createTracker` now mirrors the config-layer defense-in-depth pattern with `default: return exhaustiveTrackerConfig(config)`\n- added a local `never` helper so unexpected runtime tracker variants fail loudly instead of falling through the switch\n\nLocal gates rerun after the patch: `pnpm format:check`, `pnpm typecheck`, `pnpm lint`, and `pnpm test`.

[the remark this answers is not in the document]
Comment (jessmartin) [reply src/tracker/factory.ts:17]: Correction: the follow-up commit is `261dbc2`.

**Comment by jessmartin** [comment]
Addressed the remaining review findings in `e1fa182`:

- `createTracker` now throws `TrackerError` for the unsupported `linear` runtime path
- `requireBootstrapConfig` now throws `OrchestratorError` instead of plain `Error`

Local gates rerun successfully: `pnpm format`, `pnpm lint`, `pnpm typecheck`, `pnpm test`, and `codex review --base origin/main`.
